avm-transpiler/src/instructions.rs

```diff
@@ -32,35 +36,46 @@ impl Display for AvmInstruction {
         if let Some(indirect) = &self.indirect {
             write!(f, ", indirect: {}", indirect)?;
         }
-        // This will be either inTag or dstTag depending on the operation
-        if let Some(dst_tag) = self.tag {
-            write!(f, ", tag: {}", dst_tag as u8)?;
-        }
         if !self.operands.is_empty() {
             write!(f, ", operands: [")?;
             for operand in &self.operands {
                 write!(f, "{operand}, ")?;
             }
             write!(f, "]")?;
         };
+        // This will be either inTag or dstTag depending on the operation
+        if let Some(dst_tag) = self.tag {
+            write!(f, ", tag: {}", dst_tag as u8)?;
+        }
+        if !self.immediates.is_empty() {
+            write!(f, ", immediates: [")?;
+            for immediate in &self.immediates {
+                write!(f, "{immediate}, ")?;
+            }
+            write!(f, "]")?;
+        };
         Ok(())
     }
 }
 
 impl AvmInstruction {
     /// Bytes representation for generating AVM bytecode
+    /// Order: INDIRECT, OPERANDS, TAG, IMMEDIATES
```

Comment:
Hmm, in principle I saw no reason to introduce the immediates separately, since you could've just modified the particular opcodes and put the immediate at the end. However then I saw you have the tag in between them. Is this needed? if you do: `Indirect, addresses, immediates, tag` then I think you wouldn't need to separate the immediates and that seems conceptually right?

Or do you need the tag to be in the middle for some reason?

Reply:
@fcarreiro I had the same thoughts and the reason I nevertheless introduced the immediates is for the SET opcode, if we put the tag before the immediates then all set opcode flavors start the same which would help to have less complex relations in bytecode decomposition.

I thought that conceptually to have "immediates" is not too bad neither.

Reply:
Basically, the offset for TAG in set opcode would be at the same position (except SET_8) for all flavors. If we put TAG at the end, then we need to offset by the different immediate sizes.

Reply:
I see. Ok, not too bad.